### PR TITLE
Fix(a11y): closes #2997, add basic a11y support to section-info-option

### LINKED ISSUE #2997: Make section-info-option accessible
From https://github.com/mozilla/activity-stream/issues/2874#issuecomment-318425638

section-info-option is not accessible (for example screen/keyboard reader user will never be able to send feedback):
- it has to be in tab index
- it has also no semantic information (could be role="note")
- sr-only could be eliminated in favour title or aria-label
- it should have aria-haspopup, aria-controls and aria-expanded attributes (I would expand it as soon as it's focused)
- you can probably use aria-describedby attribute here to point at the info-option header and body
- info-option-header should have header semantics.

### PR DESCRIPTION
**Pull request by dmose**
This gets most of the suggestions in fix #2997.  We'll have to spin off a bug to do the aria-describedby work, as well as getting better focus stuff for 57 for the expanded info option.

### CODE REVIEW

File: system-addon/content-src/components/Sections/Sections.jsx [hunk 1]
```diff
@@ -1,25 +1,56 @@
...
+
   render() {
-    const {id, eventSource, title, icon, rows, infoOption, emptyState, dispatch, maxCards, contextMenuOptions} = this.props;
+    const {id, eventSource, title, icon, rows, infoOption, emptyState, dispatch, maxCards, contextMenuOptions, intl} = this.props;
     const initialized = rows && rows.length > 0;
     const shouldShowTopics = (id === "TopStories" && this.props.topics && this.props.read_more_endpoint);
+
+    const infoOptionIconA11yAttrs = {
+      "aria-haspopup": "true",
+      "aria-controls": "info-option",
```
Comment (yzen) [inline system-addon/content-src/components/Sections/Sections.jsx]: Ah so the unfortunate part of the aria-spec (IMO) is that attributes for relations such as aria-controls, aria-describedby etc only take a list of 1 or more id refs. Class names unfortunately do not work

**Comment by yzen** [comment]
Sorry if I missed it, I think one thing that's missing is a handler of a focus event that the icon would receive (when keyboard user tabs to it). I guess, we would want to do that same expansion that happens on hover. Next tab should focus on the link ensuring that the info-section does not get collapsed. The tab away from a link should probably collapse it?

File: system-addon/content-src/components/Sections/Sections.jsx (hunk 1 above)
Comment (dmose) [inline system-addon/content-src/components/Sections/Sections.jsx]: yeah, there appears to be some workarounds for this in react, i'll file a spin-off bug to deal with that by 57.

**Comment by dmose** [comment]
@yzen The focus stuff won't get done tonight.  I can either land this patch as-is/with minimal tweaks for 56, or wave it off until 57.

**Comment by Mardak** [comment]
activity-stream info tabfocus

**Comment by Mardak** [comment]
Github doesn't like my webm so here's a tiny gif..
It's hard to tell but I click Pocket text, tab to info, tab to feedback, tab past to story, shift-tab to info, tab to feedback, tab to story, shift-tab to info, shift-tab to twitter context menu.
screenflow webmhd

File: system-addon/test/unit/content-src/components/Sections.test.jsx [hunk 2]
```diff
@@ -12,14 +14,82 @@ describe("<Sections>", () => {
...
+    assert.lengthOf(wrapper.find('.info-option-icon[aria-expanded="true"]'), 1);
+  });
+
+  it('should render info-option-icon w/aria-expanded["false"] when moused out', () => {
+    const wrapper = shallowWithIntl(<Section {...FAKE_SECTION} />);
+    wrapper.find(".section-info-option").simulate("mouseover");
+
+    wrapper.find(".section-info-option").simulate("mouseout");
+
+    assert.lengthOf(wrapper.find('.info-option-icon[aria-expanded="false"]'), 1);
```
Comment (Mardak) [inline system-addon/test/unit/content-src/components/Sections.test.jsx]: Thanks for the tests! Made it easy to refactor!

**Comment by as-pine-proxy** [comment]
[Treeherder: 9b5b1cc94409ef003cd1d263e5fe815e14374c8b](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=9b5b1cc94409ef003cd1d263e5fe815e14374c8b)

**Pull request merged**